provider/source/src/datetime/symbols.rs

```diff
@@ -31,24 +36,30 @@ pub(crate) fn convert_times(other: &cldr_serde::ca::Dates) -> TimeSymbols<'stati
     }
 }
 
-fn convert_eras(eras: &cldr_serde::ca::Eras, calendar: &str) -> Eras<'static> {
+fn convert_eras(in_eras: &cldr_serde::ca::Eras, all_eras: &[(usize, EraData)]) -> Eras<'static> {
     let mut out_eras = Eras::default();
 
-    for &(cldr, code) in &get_era_code_map()[calendar] {
-        if let Some(name) = eras.names.get(&cldr.to_string()) {
-            out_eras.names.insert(code.as_str().into(), name);
+    for (index, era) in all_eras {
```

Comment:
This assumes that CLDR era indices are the same as ICU4X era indices. I would _at least_ like to have that be tested, probably with code similar to the era map below.

ICU4X formatting era indices are hardcoded.